lektor/markdown.py

```diff
@@ -11,9 +11,18 @@
 
 _markdown_cache = threading.local()
 
+_old_mistune = int(mistune.__version__.split('.', maxsplit=1)[0]) < 2
 
-class ImprovedRenderer(mistune.Renderer):
-    def link(self, link, title, text):
+if _old_mistune:
+    from mistune import Renderer as BaseRenderer
+else:
+    from mistune import HTMLRenderer as BaseRenderer  # pylint: disable=no-name-in-module
+
+
+class ImprovedRenderer(BaseRenderer):
+    def link(self, link, text=None, title=None):  # pylint: disable=arguments-differ, arguments-renamed
+        if _old_mistune:
+            (title, text) = (text, title)
```

Comment:
This swapping of arguments makes me nervous.  I think it's true that mistune always calls .link with all positional arguments (no named arguments), but that is not clear from inspection.  `title, text = text, title` doesn't really defuse the confusion.

It would be nice to keep the call signatures of the methods identical to those of the underlying renderer.

A couple of alternative ideas:

My preference, I think (at least, right now) would be something like:
```py
if _old_mistune:
    class ImprovedRenderer(mistune.Renderer):
        def link(self, link, title, text):
            return _render_link(link, text, title, record=self.record)
        def image(self, src, title, text):
            return _render_image(src, text, title, record=self.record)
else:
    class ImprovedRenderer(mistune.HTMLRenderer):
        def link(self, link, text=None, title=None):
            return _render_link(link text, title, record=self.record)
        def image(self, src, alt="", title=None):
            return _render_image(src, alt, title, record=self.record) 
        ...

def _render_link(link, text=None, title=None, record=None):
    if record is not None:
        ...


def _render_image(src, alt="", title=None, record=None):
    ...
```

A more compact — but I think more half-assed — alternative would be something like: 
```py
class ImprovedRenderer(BaseRenderer):
    def _link(self, link, text=None, title=None):
        if self.record is not None:
            ...

    if _old_mistune:
        def link(self, link, title, text):
            return self._link(link, text, title)
    else:
        link = _link
```